sklearn/manifold/spectral_embedding_.py

```diff
@@ -288,6 +288,7 @@ def spectral_embedding(adjacency, n_components=8, eigen_solver=None,
         laplacian = check_array(laplacian, dtype=np.float64,
                                 accept_sparse=True)
         laplacian = _set_diag(laplacian, 1, norm_laplacian)
+        laplacian = laplacian + 1e-5 * sparse.eye(laplacian.shape[0])
```

Comment:
This can use +=. However, could it just be performed in the previous line which also appears to focus on setting the diagonal?

Reply:
Simplified the laplacian in the latest commits

Reply:
why is it not using `+=`?
```suggestion
        laplacian += 1e-5 * sparse.eye(laplacian.shape[0])
```